### PR TITLE
Force combine_hash to produce int values

### PR DESCRIPTION
**Pull request by erichwang**

### CODE REVIEW

**Comment by martint** [comment]
Explain what's the issue in the commit message

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/CombineHashFunction.java [hunk 1]
```diff
@@ -24,6 +24,6 @@ private CombineHashFunction() {}
     @SqlType(StandardTypes.BIGINT)
     public static long getHash(@SqlType(StandardTypes.BIGINT) long previousHashValue, @SqlType(StandardTypes.BIGINT) long value)
     {
-        return 31 * previousHashValue + value;
+        return (int) (31 * previousHashValue + value);
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/CombineHashFunction.java]: Add a comment explaining why a function that returns a `long` truncates the value to an `int`.

**Comment by cberner** [comment]
Looks good, except David and Martin's comments

File: presto-docs/src/main/sphinx/release/release-0.128.rst [hunk 2]
```diff
@@ -10,3 +10,7 @@ General Changes
 * Fix Example HTTP connector.
   It would previously fail with a JSON deserialization error.
 * Optimize memory usage in TupleDomain.
+* Fix an issue that can occur when an ``INNER JOIN`` has equi-join clauses that
+  align with the grouping columns used by a preceding operation such as
+  ``GROUP BY``, ``DISTINCT``, etc. When this triggers, the join may erroneously
+  produce zero output rows.
```
Comment (cberner) [inline presto-docs/src/main/sphinx/release/release-0.128.rst]: Is it always zero, or could it be something else that's also wrong?

**Comment by cberner** [comment]
Looks good still, but see comment

**Pull request merged**